### DISCUSSION TITLE
Service_name support proxy parameter,e.g -Dskywalking.agent.service_name=${spring.application.name}

### DISCUSSION THREAD

**Discussion by lusains** [apache/skywalking#10496] [General]
When I deloy a springcloud application. I don't know the real service name. I want to use proxy parameter ${spring. Cloud. Nacos. Service.name}, is the agent support?

**Comment by wu-sheng** [comment]
This is your service, you don't know the name?

**Reply by lusains** [reply]
I'm a platform operator，from platform wise, I don't know until the app runs and registers in the registry.

**Reply by lusains** [reply]
On the platform we call it the applicaiton, I have the application name, but on the skywalking-ui it's servcie, which doesn't correspond conceptually, so I want to pass the microservice name.

**Reply by wu-sheng** [reply]
Ask for your dev team. This is not an issue we could resolve. All Spring service has relative configurations to declare service name.

**Reply by wu-sheng** [reply]
If you want to control the service name through system environment, we have provided that. Don't mix that with system properties.

**Reply by lusains** [reply]
Ok, thanks for your advice.
